### PR TITLE
feat: add swap fee to FeeDetailsBottomSheet

### PR DESCRIPTION
**Pull request by MuckT**
### Description

Adds the swap fee if available to the FeedDetailsBottomSheet used in `EarnEnterAmount.tsx`.

| iOS w Swap | Android w Swap | iOS w/o Swap | Android w/o Swap |
| ----- | ----- | ----- | ----- |
|  |  |  |  |

### Test plan

- [x] Unit tests updated
- [x] Tested locally on iOS
- [x] Tested locally on Android

### Related issues

- Part of ACT-1357

### Backwards compatibility

Yes

### Network scalability

Yes

### CODE REVIEW

File: locales/base/translation.json [hunk 1]
```diff
@@ -2571,7 +2571,9 @@
         "moreInformation": "More information",
         "estNetworkFee": "Est. Network Fee",
         "maxNetworkFee": "Max Network Fee",
-        "networkFeeDescription": "The network fee is required by the network to process the deposit transaction."
+        "networkFeeDescription": "The network fee is required by the network to process the deposit transaction.",
+        "networkSwapFeeDescription": "The network fee is required by the network to process the deposit transactions. The {{appName}} fee of {{appFeePercentage}}% is charged for your use of our product.",
+        "appSwapFee": "{{appName}} Swap Fee"
```
Comment (satish-ravi) [inline locales/base/translation.json:2576]:
```suggestion
        "appSwapFee": "{{appName}} Fee"
```
per design

File: src/earn/EarnEnterAmount.tsx [hunk 2]
```diff
@@ -659,14 +687,40 @@ function FeeDetailsBottomSheet({
             )}
           </View>
         </View>
-        <RowDivider style={{ marginBottom: Spacing.Large32 }} />
-        <View style={styles.gap8}>
+        <RowDivider />
```
Comment (satish-ravi) [inline src/earn/EarnEnterAmount.tsx:690]: nit: can we also remove the style prop from RowDivider now that it's not used? Would prefer to keep overrides at a minimum where possible

File: src/earn/EarnEnterAmount.tsx [hunk 3]
```diff
@@ -605,15 +608,40 @@ function FeeDetailsBottomSheet({
...
+  if (!depositToken) {
+    // should never happen
+    throw new Error(`Token info not found for token ID ${pool.dataProps.depositTokenId}`)
+  }
+
+  const swapFeeAmount = useMemo(() => {
+    if (swapTransaction && swapTransaction.appFeePercentageIncludedInPrice) {
+      return new BigNumber(swapTransaction.buyAmount)
+        .multipliedBy(new BigNumber(swapTransaction.appFeePercentageIncludedInPrice).shiftedBy(-2)) // To convert from percentage to decimal
+        .shiftedBy(-depositToken.decimals)
```
Comment (satish-ravi) [inline src/earn/EarnEnterAmount.tsx:637]: the app fee is a % of the from amount, not the to amount. So this should be using the amount the user inputs

File: src/earn/EarnEnterAmount.tsx [hunk 4]
```diff
@@ -659,14 +687,40 @@ function FeeDetailsBottomSheet({
...
+          <View style={styles.bottomSheetLineItem} testID="SwapFee">
+            <Text style={styles.bottomSheetLineLabel}>
+              {t('earnFlow.enterAmount.feeBottomSheet.appSwapFee')}
+            </Text>
+            <Text style={styles.bottomSheetLineLabelText} testID="SwapFee/Value">
+              {'≈ '}
+              <TokenDisplay tokenId={token.tokenId} amount={swapFeeAmount.toString()} />
+              {' ('}
+              <TokenDisplay
+                tokenId={depositToken.tokenId}
```
Comment (satish-ravi) [inline src/earn/EarnEnterAmount.tsx:701]: related to above comment, this should be the swap from token

**Review by satish-ravi — CHANGES_REQUESTED** [review]

File: src/earn/EarnEnterAmount.tsx (hunk 2 above)
Comment (MuckT) [reply src/earn/EarnEnterAmount.tsx:682]: Removed in d6d56d3139fe51ddf5661ffbb870b4399b1d49a2!

File: src/earn/EarnEnterAmount.tsx (hunk 4 above)
Comment (MuckT) [reply src/earn/EarnEnterAmount.tsx]: Fixed in fff206b091c9f59a4a2c2ea84e6cf6e7a7469de2!

File: src/earn/EarnEnterAmount.tsx (hunk 3 above)
Comment (MuckT) [reply src/earn/EarnEnterAmount.tsx]: Fixed in https://github.com/valora-inc/wallet/commit/fff206b091c9f59a4a2c2ea84e6cf6e7a7469de2!

File: src/earn/EarnEnterAmount.tsx [hunk 5]
```diff
@@ -579,15 +582,40 @@ function FeeDetailsBottomSheet({
...
+  const depositToken = useTokenInfo(pool.dataProps.depositTokenId)
+
+  if (!depositToken) {
+    // should never happen
+    throw new Error(`Token info not found for token ID ${pool.dataProps.depositTokenId}`)
+  }
+
+  const swapFeeAmount = useMemo(() => {
+    if (swapTransaction && swapTransaction.appFeePercentageIncludedInPrice) {
+      return new BigNumber(swapTransaction.buyAmount)
```
Comment (satish-ravi) [inline src/earn/EarnEnterAmount.tsx:609]: `buyAmount` is the swap to amount. Can we use the user inputted amount (tokenAmount from the parent component)?

Comment (MuckT) [reply src/earn/EarnEnterAmount.tsx:609]: This is done in #6026, but I'll add those changes here.

Comment (MuckT) [reply src/earn/EarnEnterAmount.tsx]: Fixed in 4431acd89efe946232c434c2c51bb5d0482fe57a!

**Pull request merged**